docs/migration.md

```diff
@@ -2,96 +2,253 @@
+
+### Changes to `pydantic.Field`
+
+`Field` no longer supports arbitrary keyword arguments to be added to the JSON schema. Instead, any extra
+data you want to add to the JSON schema should be passed as a dictionary to the `json_schema_extra` keyword argument.
+
+The following properties have been removed from `Field`:
+
+- `const`
+- `min_items`
+- `max_items`
+- `unique_items`
+- `allow_mutation`
+- `regex`
+
+* [TODO: Need to document all other backwards-incompatible changes to `pydantic.Field`]
+
+
+### Changes to dataclasses
+
+* When used as fields, dataclasses (Pydantic or vanilla) no longer accept tuples as validation inputs; dicts should be
+  used instead.
+* The `__post_init__` in Pydantic dataclasses will now be called _after_ validation, rather than before.
+    * As a result, the `__post_init_post_parse__` method would have become redundant, so has been removed.
+* We no longer support `extra='allow'` for Pydantic dataclasses, where extra fields passed to the initializer would be
+  stored as extra attributes on the dataclass. `extra='ignore'` is still supported for the purpose of ignoring
+  unexpected fields while parsing data, they just won't be stored on the instance.
+* Pydantic dataclasses no longer have an attribute `__pydantic_model__`, and no longer use an underlying `BaseModel`
+  to perform validation or provide other functionality.
+    * To perform validation, generate a JSON schema, or make use of
+      any other functionality that may have required `__pydantic_model__` in V1, you should now wrap the dataclass
+      with a `TypeAdapter` (discussed more below) and make use of its methods.
+    * [TODO: Add link to TypeAdapter documentation. You can find example usage in `tests/test_type_adapter.py`.]
+* In Pydantic V1, if you used a vanilla (i.e., non-Pydantic) dataclass as a field, the config of the parent type would
+  be used as though it was the config for the dataclass itself as well. In Pydantic V2, this is no longer the case.
+    * [TODO: Need to specify how to override the config used for vanilla dataclass; possibly need to add functionality?]
+
+
+### Changes to config
+
+* In Pydantic V2, to specify config on a model, you should set a class attribute called `model_config` to be a dict
+  with the key/value pairs you want to be used as the config. The Pydantic V1 behavior to create a class called `Config`
+  in the namespace of the parent `BaseModel` subclass is now deprecated.
+* The following config settings have been removed:
+    * `allow_mutation`.
+    * `error_msg_templates`.
+    * `fields` — this was the source of various bugs, so has been removed.
+      You should be able to use `Annotated` on fields to modify them as desired.
+    * `getter_dict` — `orm_mode` has been removed, and this implementation detail is no longer necessary.
+    * `schema_extra` — you should now use the `json_schema_extra` keyword argument to `pydantic.Field`.
```

Comment:
What's the difference between `schema_extra` and `json_schema_extra`? If it's a renaming, it should be on the next bullet list, not this one. 🤔